src/biokbase/narrative/tests/test_job.py

```diff
@@ -400,11 +402,73 @@ def test_job_update__invalid_job_id(self):
         """
         job = create_job_from_ee2(JOB_RUNNING)
         expected = create_state_from_ee2(JOB_RUNNING)
-        self.assertEqual(job.state(), expected)
+        self.assertEqual(job.refresh_state(), expected)
 
         # try to update it with the job state from a different job
-        with self.assertRaisesRegex(ValueError, "Job ID mismatch in _update_state"):
-            job._update_state(get_test_job(JOB_COMPLETED))
+        with self.assertRaisesRegex(ValueError, "Job ID mismatch in update_state"):
+            job.update_state(get_test_job(JOB_COMPLETED))
+
+    @mock.patch(CLIENTS, get_mock_client)
+    def test_job_update__last_updated__no_change(self):
+        for job_id, job in get_all_jobs().items():
+
+            last_updated = job.last_updated
+
+            # job has full ee2 state
+            ee2_state = get_test_job(job_id)
+            job._acc_state = get_test_job(job_id)
+
+            job.update_state(ee2_state)
+            self.assertEqual(last_updated, job.last_updated)
+
+            trim_ee2_state(ee2_state, JOB_INIT_EXCLUDED_JOB_STATE_FIELDS)
+            job.update_state(ee2_state)
+            self.assertEqual(last_updated, job.last_updated)
+
+            trim_ee2_state(ee2_state, EXCLUDED_JOB_STATE_FIELDS)
+            job.update_state(ee2_state)
+            self.assertEqual(last_updated, job.last_updated)
+
+            trim_ee2_state(ee2_state, OUTPUT_STATE_EXCLUDED_JOB_STATE_FIELDS)
+            job.update_state(ee2_state)
+            self.assertEqual(last_updated, job.last_updated)
+
+            job.update_state({})
+            self.assertEqual(last_updated, job.last_updated)
+
+            # job has init ee2 state
```

Comment:
This test looks suspiciously spaghetti code-like. Does it need to be done as this long series of transitions or can it be split into separate tests?

Reply:
I thought it followed a very similar pattern throughout and so could flow in a single function. The punchline is `last_updated` defined at the top never changes throughout these tests. Is there a benefit to making tests methods small?

Reply:
It's much easier to read, understand, and update/edit a couple of stanzas of code than it is a long series of stanzas. Unless there is a specific need to test a sequence of modifications (e.g. there's something going on elsewhere that changes state as a result of these mods), it's best to make tests as simple as possible to assist future codebase editors and maintainers.

Reply:
Ah. But what if it's two long stanzas of highly repetitive code? With a common punchline that is accentuated by more repetition?

Reply:
If it's highly repetitive, it suggests that the repetition could be abstracted out into a function... or that it could be replaced by individual tests that validate the atomic operations involved.